### PR TITLE
[Improvement](pipeline) support send eos on local exchange and remove some unused code

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
support send eos on local exchange and remove some unused code
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.84 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17164139361 Bytes

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.19 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          30.0 seconds inserted 10000000 Rows, about 333K ops/s
 storage size: 17167040153 Bytes

**Review by xinyiZzz — APPROVED** [review]
LGTM

File: be/src/vec/sink/vdata_stream_sender.cpp [hunk 1]
```diff
@@ -266,7 +266,13 @@ Status Channel::close_internal() {
         status = send_current_block(true);
     } else {
         SCOPED_CONSUME_MEM_TRACKER(_parent->_mem_tracker.get());
-        status = send_block((PBlock*)nullptr, true);
+        if (is_local()) {
+            if (_recvr_is_valid()) {
```
Comment (yiguolei) [inline be/src/vec/sink/vdata_stream_sender.cpp:270]: if just remove the sender not send a null block to receiver. Then the receiver may not get eof signal?

Comment (HappenLee) [reply be/src/vec/sink/vdata_stream_sender.cpp:270]: The receiver just judge the eof by `sender_number == 0`;

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**